### PR TITLE
sealed internal class

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/microsoft/reverse-proxy/issues/924

### CODE REVIEW

**Review by karelz — APPROVED** [review]

File: src/OperatorFramework/src/Controller/Informers/ResourceInformer.cs [hunk 1]
```diff
@@ -436,7 +436,7 @@ public Registration(ResourceInformer<TResource> resourceInformer, ResourceInform
 
             public Task ReadyAsync(CancellationToken cancellationToken) => ResourceInformer.ReadyAsync(cancellationToken);
 
-            protected virtual void Dispose(bool disposing)
+            private void Dispose(bool disposing)
```
Comment (Kahbazi) [inline src/OperatorFramework/src/Controller/Informers/ResourceInformer.cs:439]: Should I revert the changes of this class or change the Dispose method so it is only one method?

Comment (karelz) [reply src/OperatorFramework/src/Controller/Informers/ResourceInformer.cs:439]: I would revert it. OperatorFramework may live on its own eventually. No point in hyper-optimizing for things like these. Keeping the common dispose pattern is likely more future-proof.

**Comment by MihaZupan** [comment]
I can't find any editorconfig options for this specific case.
